### PR TITLE
added new 'create user' tutorial

### PR DESCRIPTION
**Pull request by dougmanis**
changes:
* added new 'add-a-user_dmanis.md' tutorial file
* added link to the new tutorial from the existing docs index topic
* supporting assignment 6.2: 'Write a first use or a tutorial topic'

### CODE REVIEW

**Comment by rbwatson** [comment]
Rebased PR branch to current `main` branch.

File: docs/tutorials/add-a-user_dmanis.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+---
+layout: page
+---
+
+# Tutorial: Add a user
+
+This tutorial will teach you how to add a new user to the To-Do system.
```
Comment (rbwatson) [inline docs/tutorials/add-a-user_dmanis.md:7]: You're underselling your tutorial--it's doing much more!

You should have an overview that describes the big picture, and how the procedures that follow support the overall goal.

File: docs/tutorials/add-a-user_dmanis.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+Allow 30 minutes to complete this lesson.
+
+## Prerequisites
+
+* The To-Do service is up and running. Follow [the setup procedure](../before-you-start-a-tutorial.md) to install and run the To-Do service on your local system.
+* You are comfortable working with [REST interfaces](https://restfulapi.net).
+* A REST client such as [cURL](https://curl.se), [Postman](https://www.postman.com), or [Insomnia](https://insomnia.rest). This tutorial assumes you are using cURL but use whatever REST client you like.
+
+## Steps
```
Comment (rbwatson) [inline docs/tutorials/add-a-user_dmanis.md:17]: This section would be a good place to describe the procedures that follow and how they contribute to the overall goal.

Then make each procedure an H3 (`###`) with a title that describes the procedure goal (as the following sections do).

File: docs/tutorials/add-a-user_dmanis.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    "id": 5
+    }
+    ```
+
+1. Verify that the response data matches the user details that you submitted.
+
+    If the data matches, congratulations! your new user is in the system and ready to receive tasks. Note the unique ID; you will need to know it if you want to fetch your user's details later. 
+
+
+## Fetch your user's details
```
Comment (rbwatson) [inline docs/tutorials/add-a-user_dmanis.md:57]:
```suggestion
### Fetch your new user's details 
```

File: docs/tutorials/add-a-user_dmanis.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    ```
+
+1. Verify that the response data matches the user details that you submitted.
+
+    If the data matches, congratulations! your new user is in the system and ready to receive tasks. Note the unique ID; you will need to know it if you want to fetch your user's details later. 
+
+
+## Fetch your user's details 
+
+You can fetch your user's details any time with the GET method of the 'users' resource with the unique ID set by the server.
```
Comment (rbwatson) [inline docs/tutorials/add-a-user_dmanis.md:59]:
```suggestion
You can fetch your user's details any time with the GET method of the 'users' resource and the instance's unique ID. 
```

File: docs/tutorials/add-a-user_dmanis.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+```json
+{
+  "first_name": "Mercedes",
+  "last_name": "Hernandez",
+  "email": "mercedes@example.com",
+  "id": 5
+}
+```
+
+## Fetch details for all users in the system
```
Comment (rbwatson) [inline docs/tutorials/add-a-user_dmanis.md:77]: I'm not sure how this is part of adding a new user.

This should be explained in the `Steps` section or removed.

File: docs/index.md
Comment (rbwatson) [inline docs/index.md:1]: I moved your new link to the Tutorials section of the topic.

**Review by rbwatson — CHANGES_REQUESTED** [review]
This looks very good!

See my review comments for more details.
After you have addressed the comments and the built documentation looks good, squash the commits in the PR and re-request a review so I can finish the merge.

**Comment by dougmanis** [comment]
> r

Bob I am thrashing in GitHub commit land. I might need help before I make more of a mess. 

I think what happened: 
* I think I must have pulled from branch 6.2-tutorial-dmanis branch to pick up your minor edits 
* My local Git prompted me to resolve some merge conflicts, which I did and then committed and pushed up to GitHub along with my own editorial changes 
* Now I have _merge_ commits, which cannot be squashed(?) 

I've attempted to roll back the merge commits and I'm just making a mess of more merges and commits and I'm getting lost. 

What can I do to clean up mess of commits so that I can give you one clean commit with only the editorial changes in it?

**Comment by rbwatson** [comment]
I'm sorry to hear you're having trouble.

When you think you've twisted your PR into a Gordian knot, you can try to pull your commits from the tangled PR and cherry-pick them over to a fresh branch. You'll have to create a new PR from the fresh branch, but that'll save your work and make the tangles disappear.

I think the easiest path to take at that point is:
1. In the GitHub website, update your fork.
2. In the GitHub desktop, create a new feature branch from `main`.
3. Select your PR/feature branch.
4. Select the commit(s) from your PR branch with your changes. Don't select any of the merge commits or any other commits. You can tell which commits are yours by using the compare branches option at the top of the commits in GitHub Desktop. 
5. [Cherry-pick that/those commit(s)](https://docs.github.com/en/desktop/managing-commits/cherry-picking-a-commit-in-github-desktop) from your PR branch to your new feature branch.
6. Review the changes in your new feature branch. Build and test the docs from your new feature branch.
7. After everything looks good, open a new PR from the new feature branch. Reference your old/broken PR in your PR description.
8. Update this PR with a new comment that contains the link to your new PR.
9. Let me know at this point by selecting the re-request review in the Reviewers column and I'll take it from there.

Your changes should be able to be cherry-picked from this branch to a new one without any problems.

**Comment by dougmanis** [comment]
Notes:
* This pull request is superseded by request #58 
* This request had some branching and merging issues so I cherry picked the changes to a new branch and new pull request

**Comment by rbwatson** [comment]
This PR was replaced by #58, which was merged and closed.

**Pull request closed without merging**